oauth2client/transport/httplib2.py

```diff
@@ -105,23 +103,39 @@ def new_request(uri, method='GET', body=None, headers=None,
             if body_stream_position is not None:
                 body.seek(body_stream_position)
 
-            return new_request(
+            # To iterate is human, to recurse, divine.
+            return self.request(
                 uri, method, body=body, headers=None,
-                redirections=redirections, connection_type=connection_type,
-                _credential_refresh_attempt=_credential_refresh_attempt+1)
+                redirections=redirections,
+                _credential_refresh_attempt=_credential_refresh_attempt + 1,
+                **kwargs)
+
+        return response, content
+
 
-        return resp, content
+def make_authorized_http(credentials, http, refresh_status_codes):
+    """Creates an http object that provides credentials to requests.
 
-    # Replace the request method with our own closure.
-    http.request = new_request
+    The behavior is transport-specific, but all transports will return a new
+    http object that provides credentials to requests and refreshed credentials
+    when a response in REFRESH_STATUS_CODES is received.
 
-    # Set credentials as a property of the request method.
-    http.request.credentials = credentials
+    Args:
+        credentials: An instance of
+            :cls:`~oauth2client.client.OAuth2Credentials`.
+        http: The http object to wrap.
+        refresh_status_codes: A sequence of status codes that indicate that
+            credentials should be refreshed and the request retried.
+
+    Returns:
+        A new http object that provides credentials to requests.
+    """
+    return AuthorizedHttp(http, credentials, refresh_status_codes)
```

Comment:
True, but I need the docstring, especially now that I just made _AuthorizedHttp private.